src/services/twitch/TwitchAPI/apiTypes.ts

```diff
@@ -0,0 +1,51 @@
+export type TwitchProfile = {
+  id: string;
+  login: string;
+  display_name: string;
+  type: string;
+  broadcaster_type: string;
+  description: string;
+  profile_image_url: string;
+  offline_image_url: string;
+  view_count: number;
+  email: string;
+};
+
+export type TwitchUsersReturnData = {
+  data: TwitchProfile[];
+};
+
+export type TwitchChatters = {
+  broadcaster: string[];
+  vips: string[];
+  moderators: string[];
+  staff: string[];
+  admins: string[];
+  global_mods: string[];
+  viewers: string[];
+};
+
+export type TwitchChattersReturnData = {
+  links: any;
+  chatter_count: number;
+  chatters: TwitchChatters;
+};
+
+export type TwitchStream = {
+  id: string;
+  user_id: string;
+  user_name: string;
+  game_id: string;
+  type: string;
+  title: string;
+  viewer_count: number;
+  started_at: string;
+  language: string;
+  thumbnail_url: string;
+  tag_ids: string[];
+};
+
+export type TwitchStreamsReturnData = {
+  data: TwitchStream[];
+  pagination: any; // idk what this is but twitch includes it in the response (never seen it with any data)
```

Comment:
With regards to this comment, the pagination property may contain a `cursor` that can be used to walk the streams that match the query supplied to the endpoint, if the count of streams exceed the supplied or default number of stream objects you've queried for. (Useful to iterate and get every stream in a particular game category, for example).

➡️ https://dev.twitch.tv/docs/api/reference#get-streams